portfolio/src/main/webapp/script.js

```diff
@@ -88,24 +90,57 @@ function fetchCommentsWithUserInput() {
   return false;
 }
 
+/**
+ * Fetches next batch of comments.
+ */
+function fetchNextPageOfComments() {
+  const limit = getCommentLimit();
+  const usernameFilter = getUsernameFilter();
+  const sortOrder = getCommentsSortOrder();
+  const map = new Map([['limit', limit], ['username', usernameFilter], ['order', sortOrder], ['start', commentsStart]]);
+  fetchComments(map, false);
+}
+
 /**
  * Fetches comments with the given requirements.
  */
-function fetchComments(requirements) {
+function fetchComments(requirements, cleanComments = true) {
   fetch('/comments' + getRequestParameter(requirements))
     .then(response => response.json())
     .then(json => {
+      const comments = json.comments;
       const commentsContainer = document.getElementById('comments-container');
-      commentsContainer.innerHTML = '';
-      for (const comment of json) {
+      if (cleanComments) {
+        commentsContainer.innerHTML = '';
+      }
+
+      if (comments.length == 0) {
```

Comment:
Does this mean a user will hit "Show more" and then nothing will happen on-screen except this button will go away since the request didn't return anything?

This is okay since you're still going through the walkthroughs but think about a way you could solve this. Maybe some sort of indicator in the API response that there are no more comments to be sent?